### PR TITLE
docs: add Local search; Add 'Edit this page' for 0.2.7 and next; Toke…

### PR DESCRIPTION
**Pull request by JohannesDienst-askui**
### Docusaurus update to 2.1.0
Had to update Docusaurus to the newest version because the Local Search Plugin would not want to work with it otherwise ;) 

### Local Search
Add local search plugin from easyops-cn that builds its index at build stage

### Edit this page
Enable 'Edit this page' link on all pages for version 0.2.7 and next

**We had to remove the _edit_custom_url: null_ from all the markdown files for this to work. It basically resets the edit_custom_url created by the 'Edit Page'-Plugin.**

### Link to Token Configuration in askui User Portal
Add a hint and a link to the documentation on Getting Started->Installing askui for the necessary token from user portal

### CODE REVIEW

File: docs/package.json [hunk 1]
```diff
@@ -15,16 +15,17 @@
...
     "clsx": "1.2.1",
     "prism-react-renderer": "1.3.5",
     "react": "17.0.2",
-    "react-dom": "17.0.2"
+    "react-dom": "17.0.2",
+    "@easyops-cn/docusaurus-search-local": "^0.32.0"
   },
   "devDependencies": {
-    "@docusaurus/types": "2.0.1",
+    "@docusaurus/types": "^2.1.0",
```
Comment (adi-wan-askui) [inline docs/package.json:28]: please pin the versions if possible so that installations are as similar as possible

Comment (JohannesDienst-askui) [reply docs/package.json:28]: I really do not like that npm does not pin the version when it was pinned before.
Will pay attention to that in the future!

File: docs/docs/general/02-Getting Started/getting-started.md [hunk 2]
```diff
@@ -38,4 +37,10 @@ Quick explanation of all the dependencies:
 If you don't like Jest, feel free to use another test framework, e.g., [Jasmine](https://jasmine.github.io/) or [Mocha](https://mochajs.org/).
 
 
+## Configure Access Token from our askui User Portal
+
+In order to use askui you have to create an access token at our [user portal](https://app.askui.com/).
+
+[After that please configure it as described here](../../api/06-Configuration/askui-ui-control-client.md#credentials)
```
[the remark this answers is not in the document]
Comment (JohannesDienst-askui) [reply docs/docs/general/02-Getting Started/getting-started.md:44]: Better text for sure!

What do you mean with "Like for using relative paths!"? Should that be included?

My dream scenario would be that the askui User Portal guides the user so perfectly that they do not need our docs for that. Maybe we should add another page under API->Configuration?

[the remark this answers is not in the document]
Comment (adi-wan-askui) [reply docs/docs/general/02-Getting Started/getting-started.md:44]: By "Like for using relative paths!" I mean that I like that you used relative paths.

File: docs/README.md [hunk 3]
```diff
@@ -24,6 +24,15 @@ npm run build
...
 
+### Local Search
+
+For your searchbar to work locally you have to generate the static content and then serve it locally.
+
+```shell
+npm run build
+npm run serve
+```
+
```
[the remark this answers is not in the document]
Comment (JohannesDienst-askui) [reply docs/README.md]: Put it under Local Search as a subheading.

File: docs/docs/general/02-Getting Started/getting-started.md [hunk 4]
```diff
@@ -38,4 +37,10 @@ Quick explanation of all the dependencies:
 If you don't like Jest, feel free to use another test framework, e.g., [Jasmine](https://jasmine.github.io/) or [Mocha](https://mochajs.org/).
 
 
+## Configure Access Token from our askui User Portal
```
[the remark this answers is not in the document]
Comment (JohannesDienst-askui) [reply docs/docs/general/02-Getting Started/getting-started.md]: Done

File: docs/docs/general/02-Getting Started/getting-started.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (JohannesDienst-askui) [reply docs/docs/general/02-Getting Started/getting-started.md]: Getting started would be a good candidate. Or do you prefer a top-level one under Getting started?

Any ideas on the name: Access Token, Token, User Token?

[the remark this answers is not in the document]
Comment (adi-wan-askui) [reply docs/docs/general/02-Getting Started/getting-started.md]: You decide! But I thought of a top-level `askui User Portal` section where we describe the functionality of it.

**Review by adi-wan-askui — COMMENTED** [review]
Looks good to me, only thing left is the User Portal stuff

File: docs/docs/general/02-Getting Started/getting-started.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (JohannesDienst-askui) [reply docs/docs/general/02-Getting Started/getting-started.md]: I hope I covered everything regarding _Sign Up_ and creating an access token in the new menu item _askui User Portal_

**Review by adi-wan-askui — APPROVED** [review]
Great work :) Let's merge this shizzle!

**Pull request merged**